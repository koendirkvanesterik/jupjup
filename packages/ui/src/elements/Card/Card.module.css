.card {
  @apply relative overflow-hidden;
  @apply flex flex-col;
  @apply w-full;
  @apply bg-white;
  @apply rounded-lg;
  @apply shadow;

  &:focus {
    @apply outline-none;
  }

  &__actions {
    @apply flex flex-wrap items-start gap-2;
  }

  &__body {
    @apply flex flex-auto flex-col gap-3;
    @apply p-6;

    :where(p) {
      @apply flex-grow;
    }
  }

  &__title {
    @apply flex items-center gap-2;
    @apply text-xl font-semibold leading-relaxed;
  }
}
