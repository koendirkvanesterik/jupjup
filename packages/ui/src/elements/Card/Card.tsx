import { PropsWithChildren, ReactNode } from 'react'
import styles from './Card.module.css'

type Props = PropsWithChildren<{
  actions?: ReactNode
  title?: string
}>

export const Card = ({ actions, children, title }: Props) => (
  <div className={styles.card}>
    <div className={styles['card__body']}>
      {title && <div className={styles['card__title']}>{title}</div>}
      {children}
      {actions && <div className={styles['card__actions']}>{actions}</div>}
    </div>
  </div>
)
