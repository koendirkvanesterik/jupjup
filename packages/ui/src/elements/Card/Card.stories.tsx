import { Button } from '../Button'
import { Card } from './Card'

const defaultProps = {
  actions: <Button>Button</Button>,
  children:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam tristique ornare lectus eu ultrices. Sed vehicula felis quis elementum posuere.',
  title: 'Title',
}

export const Base = () => (
  <div className="max-w-sm">
    <Card {...defaultProps} />
  </div>
)

export const Text = () => (
  <div className="max-w-sm">
    <Card {...defaultProps} actions={undefined} title={undefined} />
  </div>
)

export const Title = () => (
  <div className="max-w-sm">
    <Card {...defaultProps} actions={undefined} />
  </div>
)

export default {
  component: Card,
  title: 'Elements/Card',
}
