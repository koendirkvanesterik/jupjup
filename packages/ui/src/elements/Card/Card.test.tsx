import { render } from '@testing-library/react'
import { Card } from './Card'

it('should render as expected', () => {
  const { container, getByText } = render(<Card>Card</Card>)
  expect(container).toMatchSnapshot()
  expect(getByText('Card')).toHaveClass('card__body')
})

it('should render with title as expected', () => {
  const { container, getByText } = render(<Card title="Title" />)
  expect(container).toMatchSnapshot()
  expect(getByText('Title')).toHaveClass('card__title')
})

it('should render with actions as expected', () => {
  const { container, getByText } = render(<Card actions="Actions" />)
  expect(container).toMatchSnapshot()
  expect(getByText('Actions')).toHaveClass('card__actions')
})
