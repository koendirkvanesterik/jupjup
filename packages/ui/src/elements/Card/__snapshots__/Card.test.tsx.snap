// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render as expected 1`] = `
<div>
  <div
    class="card"
  >
    <div
      class="card__body"
    >
      Card
    </div>
  </div>
</div>
`;

exports[`should render with actions as expected 1`] = `
<div>
  <div
    class="card"
  >
    <div
      class="card__body"
    >
      <div
        class="card__actions"
      >
        Actions
      </div>
    </div>
  </div>
</div>
`;

exports[`should render with title as expected 1`] = `
<div>
  <div
    class="card"
  >
    <div
      class="card__body"
    >
      <div
        class="card__title"
      >
        Title
      </div>
    </div>
  </div>
</div>
`;
